src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.Value.cs

```diff
@@ -0,0 +1,264 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Analyzer.Utilities.PooledObjects;
+using System.Linq;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.GlobalFlowStateAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.ValueContentAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis;
+using System.Diagnostics.CodeAnalysis;
+using Analyzer.Utilities;
+using System.Diagnostics;
+
+namespace Microsoft.NetCore.Analyzers.InteropServices
+{
+    using ValueContentAnalysisResult = DataFlowAnalysisResult<ValueContentBlockAnalysisResult, ValueContentAbstractValue>;
+
+    public sealed partial class PlatformCompatabilityAnalyzer
+    {
+        private const string Browser = nameof(Browser);
+        private const string Linux = nameof(Linux);
+        private const string FreeBSD = nameof(FreeBSD);
+        private const string Android = nameof(Android);
+        private const string IOS = nameof(IOS);
+        private const string MacOS = nameof(MacOS);
+        private const string TvOS = nameof(TvOS);
+        private const string WatchOS = nameof(WatchOS);
+        private const string Windows = nameof(Windows);
```

Comment:
See comment for `SwitchPlatformName`, let's not hard code the platform set here. Instead, let's make the analyzer driving by conventions off of `OperatingSystem`.

Reply:
Same as above, let's queue this change up for after the release, leaving it hard-coded for now.